### PR TITLE
Added note to writing_plugins.rst

### LINKED ISSUE #7212: pytest_collection_modifyitems in subdir conftest recieves all tests
Given the following test setup:
```
tests
| - a
    | - __init__.py
    | - conftest.py
    | - test_foo.py  
| - __init__.py
| - test_bar.py
```

```python
# a/conftest.py
def pytest_collection_modifyitems(items):
    for item in items:
        print(item.name)
```

```python
# a/test_foo.py
def test_foo():
    pass
```

```python
# test_bar.py
def test_bar():
    pass
```

Based on the docs, I wouldn't expect to see "test_bar" be printed, but I do.

---
Debug:
```
Package            Version
------------------ -------
atomicwrites       1.4.0
attrs              19.3.0
colorama           0.4.3
importlib-metadata 1.6.0
more-itertools     8.2.0
packaging          20.3
pip                10.0.1
pluggy             0.13.1
py                 1.8.1
pyparsing          2.4.7
pytest             5.4.2
setuptools         39.0.1
six                1.14.0
wcwidth            0.1.9
zipp               3.1.0
```

### PR DESCRIPTION
**Pull request by nimeshvashistha**
fixes #7212

### CODE REVIEW

File: doc/en/writing_plugins.rst [hunk 1]
```diff
@@ -107,6 +107,9 @@ Here is how you might run it::
 
     See also: :ref:`pythonpath`.
 
+.. note::
+    Some functions should be implemented only in plugins or conftest.py files situated at the
```
Comment (nicoddemus) [inline doc/en/writing_plugins.rst:111]:
```suggestion
    Some hooks should be implemented only in plugins or conftest.py files situated at the
```

File: doc/en/writing_plugins.rst [hunk 2]
```diff
@@ -107,6 +107,9 @@ Here is how you might run it::
...
+    tests root directory due to how pytest discovers plugins during startup.
```
Comment (nicoddemus) [inline doc/en/writing_plugins.rst:112]:
```suggestion
    tests root directory due to how pytest discovers plugins during startup,
    see the documentation of each hook for details.
```

**Review by nicoddemus — CHANGES_REQUESTED** [review]

**Comment by nicoddemus** [comment]
Thanks @nimeshvashistha! Two small tweaks and LGTM! 👍

**Comment by The-Compiler** [comment]
#7212 specifically mentions the `pytest_collection_modifyitems` hook - so I guess something should be added to that hook (rather than just a general notice) as well?

**Comment by nimeshvashistha** [comment]
> #7212 specifically mentions the `pytest_collection_modifyitems` hook - so I guess something should be added to that hook (rather than just a general notice) as well?

Sure, I'm willing to contribute more, can you please guide me a bit about it?

**Comment by nicoddemus** [comment]
> Sure, I'm willing to contribute more, can you please guide me a bit about it?

I think a note under the [pytest_collection_modifyitems](https://docs.pytest.org/en/stable/reference.html#pytest.hookspec.pytest_collection_modifyitems) hook mentioning that, if implemented in `conftest.py` files, it always receives all collected items, not only those under the `conftest.py`.

**Comment by nimeshvashistha** [comment]
> > Sure, I'm willing to contribute more, can you please guide me a bit about it?
> 
> I think a note under the [pytest_collection_modifyitems](https://docs.pytest.org/en/stable/reference.html#pytest.hookspec.pytest_collection_modifyitems) hook mentioning that, if implemented in `conftest.py` files, it always receives all collected items, not only those under the `conftest.py`.

alright, I've made the changes.

File: doc/en/reference.rst [hunk 3]
```diff
@@ -688,6 +688,10 @@ items, delete or otherwise amend the test items:
 
 .. autofunction:: pytest_collection_modifyitems
 
+.. note::
+    If this hook is implemented in conftest.py files, it always receives all collected items, not only those
```
Comment (nicoddemus) [inline doc/en/reference.rst:692]:
```suggestion
    If this hook is implemented in ``conftest.py`` files, it always receives all collected items, not only those
```

File: doc/en/reference.rst [hunk 4]
```diff
@@ -688,6 +688,10 @@ items, delete or otherwise amend the test items:
...
+    under the conftest.py.
```
Comment (nicoddemus) [inline doc/en/reference.rst:693]:
```suggestion
    under the ``conftest.py`` where it is implemented.
```

**Review by nicoddemus — APPROVED** [review]
Thanks @nimeshvashistha!

(We should squash merge this PR)

**Pull request merged**